File: student-management-system/demo-student/src/main/java/com/example/demo/controller/StudentController.java
package com.example.demo.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.demo.model.Student;
import com.example.demo.service.StudentService;

/**
* Student controller with @RestController.
* 
* @author Caronline Lee
* Created on 12-05-2022
**/
@RestController
@RequestMapping("/student")
public class StudentController {
	
	@Autowired
	private StudentService studentService;
	
	@GetMapping("/list")
	public List<Student> listStudent() {
		return studentService.listStudent();
	}
	
	@GetMapping("/list/{id}")
	public Student selectStudent(@PathVariable int id) {
		return studentService.selectStudent(id);
	}

	@PostMapping("/insert")
	public Integer insertStudent(@RequestBody Student student) {
		return studentService.addStudent(student);
	}
	
	@PutMapping("/{id}/update")
	public Integer updateStudent(@RequestBody Student student) {
		return studentService.updateStudent(student);
	}
	
	@DeleteMapping("/{id}/delete")
	public Integer deleteStudent(@PathVariable int id) {
		return studentService.removeStudent(id);
	}
}

File: kotlin_learning/kotlin_librarian/src/main/kotlin/com/librarian/elibrarian/controller/StudentController.kt
package com.librarian.elibrarian.controller

import com.librarian.elibrarian.model.Student
import com.librarian.elibrarian.service.StudentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/api/student")
class StudentController(@Autowired private val studentService: StudentService) {

    @PostMapping
    fun addStudent(@RequestBody student: Student): ResponseEntity<Student> {
        return ResponseEntity.ok(studentService.create(student))
    }

    @GetMapping
    fun getList(): ResponseEntity<List<Student>> =
            ResponseEntity.ok(studentService.list())

    @GetMapping("/{id}")
    fun getById(@PathVariable id: Int): ResponseEntity<Optional<Student>> {
        return ResponseEntity.ok(studentService.getById(id))
    }

    @PutMapping("/{id}")
    fun updateStudent(@RequestBody student: Student): ResponseEntity<Student> {
        return ResponseEntity.ok(studentService.update(student))
    }

    @DeleteMapping("{id}")
    fun deleteStudent(@PathVariable id: Int): ResponseEntity<Int> {
        return ResponseEntity.ok(studentService.remove(id))
    }
}